### PR TITLE
Python 3: To replace some modules and funtions with six

### PR DESCRIPTION
**Pull request by hsmj1412**
Module: Configparser, cPickle, Queue, BaseHTTPServer, SimpleHTTPServer
        urllib, urllib2
Function: raw_input, reduce, unichr

Signed-off-by: Junxiang Li <junli@redhat.com>

### CODE REVIEW

**Comment by hsmj1412** [comment]
Due to unexpected error, replace the virttest/env_process.py and virttest/http_server.py with try and except.
Report: https://travis-ci.org/avocado-framework/avocado-vt/jobs/348058630

File: scripts/regression.py [hunk 1]
```diff
@@ -11,13 +11,15 @@
 import re
 import commands
 import warnings
-import ConfigParser
 
 import MySQLdb
 
 
+from six.moves import configparser
```
Comment (chunfuwen) [inline scripts/regression.py]: please use try .... except to import configparser instead of using it from six

File: virttest/http_server.py [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
-import urllib
 import logging
-import BaseHTTPServer
-import SimpleHTTPServer
+try:
+    from urllib.parse import unquote, urlparse
+    from http.server import HTTPServer, SimpleHTTPRequestHandler
+except ImportError:
+    from urllib import unquote
+    from urlparse import urlparse
```
Comment (chunfuwen) [inline virttest/http_server.py]: I think you need add import BaseHTTPServer and SimpleHTTPServer firstly

File: virttest/ppm_utils.py [hunk 3]
```diff
@@ -10,6 +10,10 @@
 import re
 import glob
 import logging
+
+from six.moves import reduce
```
Comment (chunfuwen) [inline virttest/ppm_utils.py]: use try ... except to import mapping reduce in python3 although it is removed in python3

Comment (chunfuwen) [reply virttest/ppm_utils.py]: on python 3 ,reduce is moving into functools module
from functools import reduce

File: virttest/staging/backports/simplejson/decoder.py [hunk 4]
```diff
@@ -173,7 +176,7 @@ def py_scanstring(s, end, encoding=None, strict=True,
                 uni2 = int(esc2, 16)
                 uni = 0x10000 + (((uni - 0xd800) << 10) | (uni2 - 0xdc00))
                 next_end += 6
-            char = unichr(uni)
+            char = six.unichr(uni)
```
Comment (chunfuwen) [inline virttest/staging/backports/simplejson/decoder.py]: Better use built_in function in python3:chr(xx)

File: virttest/staging/utils_koji.py [hunk 5]
```diff
@@ -1,12 +1,17 @@
...
 import logging
-import urllib
+try:
+    import configparser as ConfigParser
+except ImportError:
+    import ConfigParser
 
 from avocado.utils import path
 from avocado.utils import download
+from six.moves import html_parser
```
Comment (chunfuwen) [inline virttest/staging/utils_koji.py]: HTMLParser.HTMLParser rename to html.parser on python3, please use try .. except to do import

File: virttest/utils_test/qemu/__init__.py [hunk 6]
```diff
@@ -31,6 +31,9 @@
 from virttest.staging import utils_memory
 
 
+from six.moves import reduce
```
Comment (chunfuwen) [inline virttest/utils_test/qemu/__init__.py]: please use from functools import reduce on python3

File: virttest/env_process.py [hunk 7]
```diff
@@ -7,8 +7,11 @@
 import shutil
 import sys
 import copy
-import urllib2
 import multiprocessing
+try:
+    from urllib.request import ProxyHandler, build_opener, install_opener
```
Comment (kylazhang) [inline virttest/env_process.py]: should be from six.moves.urllib.request?

File: virttest/http_server.py [hunk 8]
```diff
@@ -1,13 +1,17 @@
...
+    from BaseHTTPServer import HTTPServer
+    from SimpleHTTPServer import SimpleHTTPRequestHandler
```
Comment (kylazhang) [inline virttest/http_server.py]: It should be separated?
try:
from urllib.parse import unquote, urlparse
except ...

try:
    from http.server import HTTPServer, SimpleHTTPRequestHandler
except ...

File: virttest/http_server.py (hunk 2 above)
Comment (kylazhang) [inline virttest/http_server.py]: Here only import is ok

File: virttest/staging/utils_koji.py [hunk 9]
```diff
@@ -1,13 +1,19 @@
...
 import os
 import time
 import logging
-import urllib
+try:
+    import configparser as ConfigParser
+    from html.parser import HTMLParser
+except ImportError:
+    import ConfigParser
+    from HTMLParser import HTMLParser
```
Comment (kylazhang) [inline virttest/staging/utils_koji.py]: Also should be separated?

File: virttest/env_process.py (hunk 7 above)
Comment (hsmj1412) [reply virttest/env_process.py]: There are something unexpected error, when I try to use six

Comment (kylazhang) [reply virttest/env_process.py]: ok, thanks

File: virttest/http_server.py (hunk 2 above)
Comment (kylazhang) [reply virttest/http_server.py]: ok, got it, thanks

**Pull request merged**